### ISSUE TITLE
Windows2008r2, unable to start chef-client service [v12.4.2] 

### ISSUE THREAD

**Issue by dlutsch** [chef/chef#3948]
Since the update to chef-client version 12.4.2 we are no longer able to bootstrap our windows server 2008 nodes. The chef-client windows service no longer starts and throws a system call error: "The service did not respond to the start or control request in a timely fashion". We had no issues with this on 12.4.1

```
Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation.  All rights reserved.

C:\Windows\system32>chef-client
Starting Chef Client, version 12.4.2
[2015-09-22T23:18:22+00:00] INFO: *** Chef 12.4.2 ***
[2015-09-22T23:18:22+00:00] INFO: Chef-client pid: 2708
[2015-09-22T23:18:32+00:00] INFO: Run List is [role[windows-base]]
[2015-09-22T23:18:32+00:00] INFO: Run List expands to [role-windows-base]
[2015-09-22T23:18:32+00:00] INFO: Starting Chef Run for build19.qa.cpf.com
[2015-09-22T23:18:32+00:00] INFO: Running start handlers
[2015-09-22T23:18:32+00:00] INFO: Start handlers complete.
[2015-09-22T23:18:32+00:00] INFO: HTTP Request Returned 404 Object Not Found:
resolving cookbooks for run list: ["role-windows-base"]
[2015-09-22T23:18:32+00:00] INFO: Loading cookbooks [role-windows-base@0.1.0, wi
ndows@1.37.0, chef_handler@1.1.6, webpi@1.1.0, chef-client@4.3.0, cron@1.2.4, lo
grotate@1.3.0]
Synchronizing Cookbooks:
  - role-windows-base
  - windows
  - chef_handler
  - webpi
  - chef-client
  - cron
  - logrotate
Compiling Cookbooks...
[2015-09-22T23:18:32+00:00] WARN: You are overriding windows_package on {:os=>"w
indows"} with Chef::Resource::WindowsCookbookPackage: used to be Chef::Resource:
:WindowsPackage. Use override: true if this is what you intended.
[2015-09-22T23:18:32+00:00] WARN: chef_gem[win32-api] chef_gem compile_time inst
allation is deprecated
[2015-09-22T23:18:32+00:00] WARN: chef_gem[win32-api] Please set `compile_time f
alse` on the resource to use the new behavior.
[2015-09-22T23:18:32+00:00] WARN: chef_gem[win32-api] or set `compile_time true`
 on the resource if compile_time behavior is required.
Recipe: windows::default
  * chef_gem[win32-api] action install[2015-09-22T23:18:32+00:00] INFO: Processi
ng chef_gem[win32-api] action install (windows::default line 23)
 (up to date)
[2015-09-22T23:18:34+00:00] WARN: chef_gem[win32-service] chef_gem compile_time
installation is deprecated
[2015-09-22T23:18:34+00:00] WARN: chef_gem[win32-service] Please set `compile_ti
me false` on the resource to use the new behavior.
[2015-09-22T23:18:34+00:00] WARN: chef_gem[win32-service] or set `compile_time t
rue` on the resource if compile_time behavior is required.
  * chef_gem[win32-service] action install[2015-09-22T23:18:34+00:00] INFO: Proc
essing chef_gem[win32-service] action install (windows::default line 23)
 (up to date)
[2015-09-22T23:18:35+00:00] WARN: chef_gem[windows-api] chef_gem compile_time in
stallation is deprecated
[2015-09-22T23:18:35+00:00] WARN: chef_gem[windows-api] Please set `compile_time
 false` on the resource to use the new behavior.
[2015-09-22T23:18:35+00:00] WARN: chef_gem[windows-api] or set `compile_time tru
e` on the resource if compile_time behavior is required.
  * chef_gem[windows-api] action install[2015-09-22T23:18:35+00:00] INFO: Proces
sing chef_gem[windows-api] action install (windows::default line 31)
 (up to date)
[2015-09-22T23:18:35+00:00] WARN: chef_gem[windows-pr] chef_gem compile_time ins
tallation is deprecated
[2015-09-22T23:18:35+00:00] WARN: chef_gem[windows-pr] Please set `compile_time
false` on the resource to use the new behavior.
[2015-09-22T23:18:35+00:00] WARN: chef_gem[windows-pr] or set `compile_time true
` on the resource if compile_time behavior is required.
  * chef_gem[windows-pr] action install[2015-09-22T23:18:35+00:00] INFO: Process
ing chef_gem[windows-pr] action install (windows::default line 31)
 (up to date)
[2015-09-22T23:18:35+00:00] WARN: chef_gem[win32-dir] chef_gem compile_time inst
allation is deprecated
[2015-09-22T23:18:35+00:00] WARN: chef_gem[win32-dir] Please set `compile_time f
alse` on the resource to use the new behavior.
[2015-09-22T23:18:35+00:00] WARN: chef_gem[win32-dir] or set `compile_time true`
 on the resource if compile_time behavior is required.
  * chef_gem[win32-dir] action install[2015-09-22T23:18:35+00:00] INFO: Processi
ng chef_gem[win32-dir] action install (windows::default line 31)
 (up to date)
[2015-09-22T23:18:35+00:00] WARN: chef_gem[win32-event] chef_gem compile_time in
stallation is deprecated
[2015-09-22T23:18:35+00:00] WARN: chef_gem[win32-event] Please set `compile_time
 false` on the resource to use the new behavior.
[2015-09-22T23:18:35+00:00] WARN: chef_gem[win32-event] or set `compile_time tru
e` on the resource if compile_time behavior is required.
  * chef_gem[win32-event] action install[2015-09-22T23:18:35+00:00] INFO: Proces
sing chef_gem[win32-event] action install (windows::default line 31)
 (up to date)
[2015-09-22T23:18:35+00:00] WARN: chef_gem[win32-mutex] chef_gem compile_time in
stallation is deprecated
[2015-09-22T23:18:35+00:00] WARN: chef_gem[win32-mutex] Please set `compile_time
 false` on the resource to use the new behavior.
[2015-09-22T23:18:35+00:00] WARN: chef_gem[win32-mutex] or set `compile_time tru
e` on the resource if compile_time behavior is required.
  * chef_gem[win32-mutex] action install[2015-09-22T23:18:35+00:00] INFO: Proces
sing chef_gem[win32-mutex] action install (windows::default line 31)
 (up to date)
  Converging 16 resources
  * chef_gem[win32-api] action install[2015-09-22T23:18:35+00:00] INFO: Processi
ng chef_gem[win32-api] action install (windows::default line 23)
 (up to date)
  * chef_gem[win32-service] action install[2015-09-22T23:18:36+00:00] INFO: Proc
essing chef_gem[win32-service] action install (windows::default line 23)
 (up to date)
  * chef_gem[windows-api] action install[2015-09-22T23:18:36+00:00] INFO: Proces
sing chef_gem[windows-api] action install (windows::default line 31)
 (up to date)
  * chef_gem[windows-pr] action install[2015-09-22T23:18:36+00:00] INFO: Process
ing chef_gem[windows-pr] action install (windows::default line 31)
 (up to date)
  * chef_gem[win32-dir] action install[2015-09-22T23:18:36+00:00] INFO: Processi
ng chef_gem[win32-dir] action install (windows::default line 31)
 (up to date)
  * chef_gem[win32-event] action install[2015-09-22T23:18:36+00:00] INFO: Proces
sing chef_gem[win32-event] action install (windows::default line 31)
 (up to date)
  * chef_gem[win32-mutex] action install[2015-09-22T23:18:36+00:00] INFO: Proces
sing chef_gem[win32-mutex] action install (windows::default line 31)
 (up to date)
Recipe: webpi::install-msi
  * remote_file[msi] action create[2015-09-22T23:18:36+00:00] INFO: Processing r
emote_file[msi] action create (webpi::install-msi line 26)
 (up to date)
  * windows_package[Web Platform Installer] action install[2015-09-22T23:18:37+0
0:00] INFO: Processing windows_package[Web Platform Installer] action install (w
ebpi::install-msi line 32)
[2015-09-22T23:18:37+00:00] INFO: Installing windows_package[Web Platform Instal
ler] version latest
[2015-09-22T23:18:37+00:00] INFO: Starting installation...this could take awhile
.


Recipe: chef-client::windows_service
  * directory[C:/chef/run] action create[2015-09-22T23:18:38+00:00] INFO: Proces
sing directory[C:/chef/run] action create (chef-client::windows_service line 53)

 (up to date)
  * directory[C:/chef/cache] action create[2015-09-22T23:18:38+00:00] INFO: Proc
essing directory[C:/chef/cache] action create (chef-client::windows_service line
 53)
 (up to date)
  * directory[C:/chef/backup] action create[2015-09-22T23:18:38+00:00] INFO: Pro
cessing directory[C:/chef/backup] action create (chef-client::windows_service li
ne 53)
 (up to date)
  * directory[C:/chef/log] action create[2015-09-22T23:18:38+00:00] INFO: Proces
sing directory[C:/chef/log] action create (chef-client::windows_service line 53)

 (up to date)
  * directory[C:/chef] action create[2015-09-22T23:18:38+00:00] INFO: Processing
 directory[C:/chef] action create (chef-client::windows_service line 53)
 (up to date)
  * execute[register-chef-service] action run[2015-09-22T23:18:38+00:00] INFO: P
rocessing execute[register-chef-service] action run (chef-client::windows_servic
e line 32)
 (skipped due to not_if)
  * windows_service[chef-client] action enable[2015-09-22T23:18:38+00:00] INFO:
Processing windows_service[chef-client] action enable (chef-client::windows_serv
ice line 37)
 (up to date)
  * windows_service[chef-client] action start[2015-09-22T23:18:38+00:00] INFO: P
rocessing windows_service[chef-client] action start (chef-client::windows_servic
e line 37)
[2015-09-22T23:18:38+00:00] INFO: windows_service[chef-client] configured with {
:service_name=>"chef-client"}


    ============================================================================
====
    Error executing action `start` on resource 'windows_service[chef-client]'
    ============================================================================
====

    SystemCallError
    ---------------
    The service did not respond to the start or control request in a timely fash
ion. - StartService: The service did not respond to the start or control request
 in a timely fashion.

    Resource Declaration:
    ---------------------
    # In c:/chef/cache/cookbooks/chef-client/recipes/windows_service.rb

     37: service 'chef-client' do
     38:   action [:enable, :start]
     39: end

    Compiled Resource:
    ------------------
    # Declared in c:/chef/cache/cookbooks/chef-client/recipes/windows_service.rb
:37:in `from_file'

    windows_service("chef-client") do
      action [:enable, :start]
      supports {:restart=>false, :reload=>false, :status=>false}
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      service_name "chef-client"
      enabled true
      pattern "chef-client"
      startup_type :automatic
      declared_type :service
      cookbook_name "chef-client"
      recipe_name "windows_service"
    end

[2015-09-22T23:18:38+00:00] INFO: Running queued delayed notifications before re
-raising exception

Running handlers:
[2015-09-22T23:18:38+00:00] ERROR: Running exception handlers
Running handlers complete
[2015-09-22T23:18:38+00:00] ERROR: Exception handlers complete
Chef Client failed. 1 resources updated in 21.14799 seconds
[2015-09-22T23:18:38+00:00] FATAL: Stacktrace dumped to c:/chef/cache/chef-stack
trace.out
[2015-09-22T23:18:38+00:00] FATAL: SystemCallError: windows_service[chef-client]
 (chef-client::windows_service line 37) had an error: SystemCallError: The servi
ce did not respond to the start or control request in a timely fashion. - StartS
ervice: The service did not respond to the start or control request in a timely
fashion.
```

**Comment by jaym** [comment]
How was the service installed?

@jkeiser @ksubrama @danielsdeleo this could be related to the appbundler changes

**Comment by jaym** [comment]
Btw, I reproed by running `chef-server-manager -a install`

I suspect this change is now breaking things: https://github.com/chef/chef/commit/983499e2408ef80761cf0b69c95508a388c6b1f3#diff-f87b217a535db4d5098bd418b5e6c61e

**Comment by jaym** [comment]
I don't think we can reasonably change that in a patch or minor release. We should do another release with appbundler 0.4.0 if at all possible. Otherwise we need to find another solution to whatever issue required us to update appbundler.

**Comment by jaym** [comment]
We should probably change this to do whatever the msi installer does for the next major. And of course update the chef-client cookbook to be idempotent for installing the windows service

**Comment by burnzy** [comment]
I've noticed this same issue for 2012r2 nodes with v12.4.2
I had to roll back to v12.4.1

**Comment by jaym** [comment]
yea, no windows nodes will work if they use `chef-service-manager` to install the service. The feature in the installer should work.

**Comment by jaym** [comment]
This will be fixed in chef 12.4.3
